src/OpenTelemetry.Exporter.Prometheus.AspNetCore/PrometheusExporterMiddleware.cs

```diff
@@ -102,4 +107,16 @@ public async Task InvokeAsync(HttpContext httpContext)
 
         this.exporter.OnExport = null;
     }
+
+    private static bool AcceptsOpenMetrics(HttpRequest request)
+    {
+        var acceptHeader = request.Headers.Accept;
+
+        if (StringValues.IsNullOrEmpty(acceptHeader))
+        {
+            return false;
+        }
+
+        return acceptHeader.Any(PrometheusHeadersParser.AcceptsOpenMetrics);
```

Comment:
Use `foreach` instead of `IEnumerable.Any` as the latter would cause allocation.

You could write up a simple benchmark to measure the allocation. For reference, here is a benchmark that I wrote for comparing `foreach` vs `IEnumerable.Any`

| Method  | Mean     | Error    | StdDev   | Gen0   | Allocated |
|-------- |---------:|---------:|---------:|-------:|----------:|
| Any     | 53.20 ns | 1.058 ns | 1.218 ns | 0.0102 |      64 B |
| ForEach | 22.38 ns | 0.255 ns | 0.226 ns |      - |         - |

```csharp

public class EnumerationBenchmarks
{
    private static readonly StringValues StringValues = new StringValues(new string[] { "ABC", "DEF", "XYZ" });

    [Benchmark]
    public bool Any()
    {
        return StringValues.Any(DoSomething);
    }

    [Benchmark]
    public bool ForEach()
    {
        foreach (var s in StringValues)
        {
            if (DoSomething(s))
            {
                return true;
            }
        }

        return false;
    }

    private static bool DoSomething(string input)
    {
        return input.Contains("1");
    }
}
```

Reply:
Roger that, updated